apps/site/assets/ts/projects/components/Banner.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement } from "react";
+import { ProjectTeaserWithDate, Route } from "./__projects";
+
+interface Props {
+  banner: ProjectTeaserWithDate;
+}
+
+const bannerRouteClass = ({ routes }: ProjectTeaserWithDate): string =>
+  cmsRouteToClass(routes[0]);
+
+const bannerBgClass = (banner: ProjectTeaserWithDate): string => {
+  if (banner.routes.length === 0) return "u-bg--unknown";
+  return bannerRouteClass(banner);
+};
+
+const toCSSClass = (name: string): string =>
+  name
+    .toLowerCase()
+    .replace(" ", "-")
+    .replace("_", "-");
+
+export const cmsRouteToClass = (route: Route): string => {
+  if (route.id === "Green") return "green-line";
+  if (route.id === "silver_line") return "silver-line";
+  if (route.id === "mattapan") return "mattapan";
+  if (route.group === "custom" && route.mode === null) return "unknown";
+  if (route.group === "custom") return toCSSClass(route.mode!);
+  if (route.group === "mode" && route.id !== null) return toCSSClass(route.id);
+  return "unknown";
+};
+
+const bannerContentClass = (banner: ProjectTeaserWithDate): string =>
+  `m-banner__content m-banner__content--default m-banner__content--left u-bg--${bannerBgClass(
+    banner
+  )}`;
+
+const bannerUpdated = (
+  banner: ProjectTeaserWithDate
+): ReactElement<HTMLElement> => <span>updated {banner.formatted_date}</span>;
+
+const BannerContent = ({ banner }: Props): ReactElement<HTMLElement> => (
+  <div className={bannerContentClass(banner)}>
+    <div className="m-banner__top">
+      <h2 className={`h2 m-banner__title m-banner__title--default`}>
+        {banner.title}
+      </h2>
+    </div>
+    <div className="m-banner__bottom">{bannerUpdated(banner)}</div>
+  </div>
+);
+
+const Banner = ({ banner }: Props): ReactElement<HTMLElement> => (
+  <a
+    href={banner.path}
+    className="m-banner m-banner--responsive m-banner--small-margin-top m-banner--default"
+  >
+    <div
+      className="m-banner__image m-banner__image--responsive m-banner__image--default"
+      style={{ backgroundImage: ` url(${banner.image.url})` }}
```

Comment:
It's not guaranteed that `image` will be set, we have a placeholder image that we can display if the image is missing. See how `placeholderImageUrl` is used in `MoreProjectsRow.tsx` for example.

Reply:
👍 

I'm not sure if we want to use the same placeholder for the banner but I'll use this for now. @ingridpierre